### PR TITLE
Add NewClusterPage which replaces newClusterModal

### PR DESCRIPTION
**Pull request by jtomasek**
assisted-new-cluster-page

### CODE REVIEW

File: src/components/clusters/Clusters.tsx [hunk 1]
```diff
@@ -84,13 +82,20 @@ const Clusters: React.FC<ClustersProps> = ({ history }) => {
             icon={AddCircleOIcon}
             title="Create new bare metal cluster"
             content="There are no clusters yet. This wizard is going to guide you through the OpenShift bare metal cluster deployment."
-            primaryAction={<NewClusterModalButton onClick={openModal} />}
+            primaryAction={
+              <Button
+                variant={ButtonVariant.primary}
+                component={(props) => <Link to={`${routeBasePath}/clusters/~new`} {...props} />}
```
Comment (rawagner) [inline src/components/clusters/Clusters.tsx:88]: I prefer
```suggestion
                onClick={() => history.push(`${routeBasePath}/clusters/~new`)}
```

File: src/components/clusters/Clusters.tsx [hunk 2]
```diff
@@ -103,7 +108,13 @@ const Clusters: React.FC<ClustersProps> = ({ history }) => {
           </PageSection>
           <AlertsSection />
           <ClusterToolbar>
-            <NewClusterModalButton onClick={openModal} ButtonComponent={ToolbarButton} />
+            <ToolbarButton
+              variant={ButtonVariant.primary}
+              component={(props) => <Link to={`${routeBasePath}/clusters/~new`} {...props} />}
```
Comment (rawagner) [inline src/components/clusters/Clusters.tsx:113]:
```suggestion
              onClick={() => history.push(`${routeBasePath}/clusters/~new`)}
```

File: src/components/clusters/Clusters.tsx [hunk 3]
```diff
@@ -115,12 +126,7 @@ const Clusters: React.FC<ClustersProps> = ({ history }) => {
         </>
       );
   }
-  return (
-    <>
-      {body}
-      {isModalOpen && <NewClusterModal closeModal={closeModal} history={history} />}
-    </>
-  );
+  return <>{body}</>;
```
Comment (rawagner) [inline src/components/clusters/Clusters.tsx:129]:
```suggestion
  return body;
```

or we can just return directly in switch/case statements

File: src/components/clusters/Clusters.tsx (hunk 1 above)
Comment (jtomasek) [reply src/components/clusters/Clusters.tsx]: Done

File: src/components/clusters/Clusters.tsx (hunk 2 above)
Comment (jtomasek) [reply src/components/clusters/Clusters.tsx]: Done

File: src/components/clusters/Clusters.tsx (hunk 3 above)
Comment (jtomasek) [reply src/components/clusters/Clusters.tsx]: Done

**Review by rawagner — APPROVED** [review]

**Pull request merged**